gbdxtools/s3.py

```diff
@@ -87,7 +87,10 @@ def download(self, location, local_dir='.'):
         location = location.strip('/')
 
         self.logger.debug('Downloading contents')
-        for s3key in s3conn.list_objects(Bucket=bucket, Prefix=(prefix+'/'+location))['Contents']:
+        objects = s3conn.list_objects(Bucket=bucket, Prefix=(prefix+'/'+location))
+        if 'Contents' not in objects:
```

Comment:
If 'Contents' doesnt exist it means it either actually does NOT exist or the user does NOT have permissions to see it right? 

Would be great to know if its one or the other and inform the user. But this is a step in the right dir

Reply:
I don't know about permissions on a given location - just that if your credentials are wrong you get a ClientError. Do we have a file we can test against?

Reply:
The way i was testing this was to use the s3 mod in gbdxtools to upload a file and make sure i can download it. But then also need to try to downsome known private file in s3 as well. I might take a look right now at boto and see what permissions logic it may have.

Reply:
So the purpose of this download method is to download an image inside your own bucket/prefix within the gbdx customer buckets. You should not have permissions to download something outside of the bucket, but im not sure how to make a file in that location that you cannot read.